.changelog/12817.txt

```diff
@@ -0,0 +1,6 @@
+```release-note:enhancement
+resource/aws_cloudwatch_metric_alarm: Add plan time validation to `alarm_name`, 
+`comparison_operator`, `metric_name`, `metric_query.id`, `metric_query.expression`, 
+`metric_query.metric.metric_name`, `metric_query.metric.namespace`, `metric_query.metric.unit`,
+`namespace`, `period`, `statistic`, `alarm_description`, `insufficient_data_actions`, `ok_actions`, `unit`, and `extended_statistic`
```

Comment:
Newlines should be omitted from changelog entries. Reference: https://github.com/hashicorp/terraform-provider-aws/issues/17135

```suggestion
resource/aws_cloudwatch_metric_alarm: Add plan time validation to `alarm_name`, `comparison_operator`, `metric_name`, `metric_query.id`, `metric_query.expression`, `metric_query.metric.metric_name`, `metric_query.metric.namespace`, `metric_query.metric.unit`, `namespace`, `period`, `statistic`, `alarm_description`, `insufficient_data_actions`, `ok_actions`, `unit`, and `extended_statistic`
```